### PR TITLE
feat(lambda): `function.addAlias()` simplifies Alias creation

### PR DESCRIPTION
**Pull request by rix0rrr**
Deprecate `version.addAlias()` in favor of a new method called
`function.addAlias()`.

- When Aliases get created underneath Versions, there's the risk
  that it's `currentVersion` whose logical ID changes on every
  deployment, causing a (probably failing) replacement of the `Alias`
  on every function update.
- The most common use for `function.currentVersion` is to create an
  `Alias` for a Function. Might as well take out the middle man.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda/lib/function.ts [hunk 1]
```diff
@@ -959,6 +961,31 @@ export class Function extends FunctionBase {
...
+   * the function as it is being updated during a deployment.
+   *
+   * ```ts
+   * declare const fn: lambda.Function;
+   *
+   * fn.addAlias('Live');
+   *
+   * // Is equivalent to
+   *
+   * new Alias(this, 'AliasLive', {
```
Comment (kaizen3031593) [inline packages/@aws-cdk/aws-lambda/lib/function.ts:977]:
```suggestion
   * new lambda.Alias(this, 'AliasLive', {
```

I'm anticipating this breaking rosetta the way it is written.

File: packages/@aws-cdk/aws-lambda/test/function.test.ts [hunk 2]
```diff
@@ -547,6 +547,18 @@ describe('function', () => {
         Annotations.fromStack(stack).hasNoWarning('/Default/MyLambda/$LATEST', Match.stringLikeRegexp(warningMessage));
       });
 
+      test('function.addAlias', () => {
```
Comment (kaizen3031593) [inline packages/@aws-cdk/aws-lambda/test/function.test.ts:550]: If I'm understanding correctly, the issue with `fn.currentVersion.addAlias()` is that the Alias will be replaced on every function update. I think we should test that `fn.addAlias()` solves the issue. This test doesn't really tell me anything.

To be honest, I'm not clear how `function.addAlias()` changes this behavior. To me, it seems like both `fn.addAlias()` and `fn.currentVersion.addAlias()` results in the same thing: `return addAlias(this, <current-version>, aliasName, options)`.

**Review by kaizen3031593 — CHANGES_REQUESTED** [review]
> When Aliases get created underneath Versions, there's the risk
    that it's currentVersion whose logical ID changes on every
    deployment, causing a (probably failing) replacement of the Alias
    on every function update.
    The most common use for function.currentVersion is to create an
    Alias for a Function. Might as well take out the middle man.

I'm a bit confused here -- see my inline comments below. The above reads
like the common use case for `fn.currentVersion.addAlias()` will cause 
potentially failing replacement of the Alias on every function update. If so,
I'm unclear how the code changes solves this -- I trust you, I just need it
to be pointed out :).

File: packages/@aws-cdk/aws-lambda/test/function.test.ts (hunk 2 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-lambda/test/function.test.ts:550]: Yes, but the `this` is different! In one case it's the `Function` and the other it's the `Version`.

But I think I was mistaken anyway: I thought the `Alias` would be replaced if the `Version` object changed... but I'm now pretty convinced it doesn't.

Comment (rix0rrr) [reply packages/@aws-cdk/aws-lambda/test/function.test.ts:550]: So this is just convenience.

**Review by kaizencc — APPROVED** [review]

**Pull request merged**